ml-agents/mlagents/trainers/policy/tf_policy.py

```diff
@@ -392,18 +399,52 @@ def get_update_vars(self):
         """
         return list(self.update_dict.keys())
 
-    def save_model(self, steps):
+    def checkpoint(self, model_reward: Optional[float] = None) -> None:
         """
-        Saves the model
-        :param steps: The number of steps the model was trained for
-        :return:
+        Writes an intermediate checkpoint model to memory
+        model_reward: Mean reward of the reward buffer at the time of saving
         """
+        current_step = self.get_current_step()
         with self.graph.as_default():
-            last_checkpoint = os.path.join(self.model_path, f"model-{steps}.ckpt")
-            self.saver.save(self.sess, last_checkpoint)
+            last_checkpoint = os.path.join(
+                self.model_path, f"model-{current_step}.ckpt"
+            )
+            if self.saver:
+                self.saver.save(self.sess, last_checkpoint)
```

Comment:
It kind of bothers me that saving the checkpoint is not handled by the CheckpointManager. Can we try to put this logic in the CheckpointManager to make this code nicer? The CheckpointManager is already responsible for deleting those files, and I think classes should delete only what they create.

Reply:
I think maybe we view `CheckpointManager` differently.  I view that as managing our own .nn checkpoints rather than implementation specific (e.g. TF/Torch) policy checkpoints.  If we want to combine those we could make `CheckpointManager` non-static, so we can have a `TFCheckpointManager` that uses the saver.  This aligns with @ervteng's suggestion that the Policy not know much about steps/rewards.

Does that make sense?  I think it's not a bad way to go moving forward, but maybe in a follow-up PR?

Reply:
Checkpoint manager should be renamed IMO. 
 - NNFileManager
 - ModelMetaData
 - NNInventory